routes/stake.go

```diff
@@ -0,0 +1,417 @@
+package routes
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"github.com/deso-protocol/core/lib"
+	"github.com/holiman/uint256"
+	"io"
+	"net/http"
+)
+
+type StakeRewardMethod string
+
+const (
+	PayToBalance StakeRewardMethod = "PAY_TO_BALANCE"
+	Restake      StakeRewardMethod = "RESTAKE"
+)
+
+type StakeRequest struct {
+	TransactorPublicKeyBase58Check string            `safeForLogging:"true"`
+	ValidatorPublicKeyBase58Check  string            `safeForLogging:"true"`
+	RewardMethod                   StakeRewardMethod `safeForLogging:"true"`
+	StakeAmountNanos               *uint256.Int      `safeForLogging:"true"`
+	ExtraData                      map[string]string `safeForLogging:"true"`
+	MinFeeRateNanosPerKB           uint64            `safeForLogging:"true"`
+	TransactionFees                []TransactionFee  `safeForLogging:"true"`
+}
+
+type UnstakeRequest struct {
+	TransactorPublicKeyBase58Check string            `safeForLogging:"true"`
+	ValidatorPublicKeyBase58Check  string            `safeForLogging:"true"`
+	UnstakeAmountNanos             *uint256.Int      `safeForLogging:"true"`
+	ExtraData                      map[string]string `safeForLogging:"true"`
+	MinFeeRateNanosPerKB           uint64            `safeForLogging:"true"`
+	TransactionFees                []TransactionFee  `safeForLogging:"true"`
+}
+
+type UnlockStakeRequest struct {
+	TransactorPublicKeyBase58Check string            `safeForLogging:"true"`
+	ValidatorPublicKeyBase58Check  string            `safeForLogging:"true"`
+	StartEpochNumber               uint64            `safeForLogging:"true"`
+	EndEpochNumber                 uint64            `safeForLogging:"true"`
+	ExtraData                      map[string]string `safeForLogging:"true"`
+	MinFeeRateNanosPerKB           uint64            `safeForLogging:"true"`
+	TransactionFees                []TransactionFee  `safeForLogging:"true"`
+}
+
+type StakeTxnResponse struct {
+	SpendAmountNanos  uint64
+	TotalInputNanos   uint64
+	ChangeAmountNanos uint64
+	FeeNanos          uint64
+	Transaction       *lib.MsgDeSoTxn
+	TransactionHex    string
+	TxnHashHex        string
+}
+
+type StakeEntryResponse struct {
+	StakerPublicKeyBase58Check    string
+	ValidatorPublicKeyBase58Check string
+	RewardMethod                  StakeRewardMethod
+	StakeAmountNanos              *uint256.Int
+	ExtraData                     map[string]string
+}
+
+type LockedStakeEntryResponse struct {
+	StakerPublicKeyBase58Check    string
+	ValidatorPublicKeyBase58Check string
+	LockedAmountNanos             *uint256.Int
+	LockedAtEpochNumber           uint64
+	ExtraData                     map[string]string
+}
+
+func (fes *APIServer) Stake(ww http.ResponseWriter, req *http.Request) {
```

Comment:
Is this a more representative name?
```suggestion
func (fes *APIServer) CreateStakeTxn(ww http.ResponseWriter, req *http.Request) {
```